python/paddle/fluid/incubate/fleet/collective/__init__.py

```diff
@@ -249,7 +222,67 @@ def _transpile(self, startup_program, main_program):
             program=main_program,
             current_endpoint=current_endpoint)
 
+    def _get_node_ips_from_endpoints(self, endpoints):
+        ss = set()
+        ips = []
+        for ep in endpoints:
+            ip = ep.split(":")[0].strip()
+            if ip not in ss:
+                ss.add(ip)
+                ips.append(ip)
+            else:
+                continue
+
+        return ips
+
+    def _node_num(self):
+        worker_endpoints = fleet.worker_endpoints()
+        current_endpoint = fleet.worker_endpoints()[fleet.worker_index()]
+        worker_endpoints_env = ','.join(worker_endpoints)
+
+        node_ips = self._get_node_ips_from_endpoints(worker_endpoints)
+        node_ip = current_endpoint.split(":")[0].strip()
+
+        node_num = len(node_ips)
+
+        return node_num
+
     def _try_to_compile(self, startup_program, main_program):
+        node_num = self._node_num()
+        assert node_num >= 1, "nccl2 node_num must >= 1, now:{}" % node_num
+
+        self._strategy.fuse_all_reduce_ops = True
+        exec_strategy = self._strategy.exec_strategy
+
+        if node_num <= 1:
+            self._strategy.nccl_comm_num = 1
+            if self._strategy.nccl_comm_num > 1:
+                logging.warn("set nccl_comm_num=1 since you only have 1 node.")
```

Comment:
self._strategy.nccl_comm_num = 1 may put the end of the condition.

Reply:
Thanks! Done!